### PR TITLE
Foreskin

### PR DESCRIPTION
**Pull request by Acensti**
## Changelog
:cl:
add: Added new things
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by nednaZ** [comment]
Are you...are you okay?

File: code/__DEFINES/jobs.dm [hunk 1]
```diff
@@ -45,3 +45,4 @@
 #define JOB_DISPLAY_ORDER_DETECTIVE 33
 #define JOB_DISPLAY_ORDER_SECURITY_OFFICER 34
 #define JOB_DISPLAY_ORDER_PRISONER 35
+#define JOB_DISPLAY_ORDER_RABBI 36
```
Comment (nednaZ) [inline code/__DEFINES/jobs.dm:48]: This should probably be 16, as it is related to the Chaplain.
Shift the rest up to match.

File: code/modules/jobs/job_types/chaplain.dm [hunk 2]
```diff
@@ -104,7 +104,7 @@
 		if("toolboxia","greytide")
 			B.name = pick("Toolbox Manifesto","iGlove Assistants")
 		if("weeaboo","kawaii")
-			B.name = pick("Fanfiction Compendium","Japanese for Dummies","The Manganomicon","Establishing Your O.T.P")
+			B.name = pick("Disgusting furry fags", "Fanfiction Compendium","Japanese for Dummies","The Manganomicon","Establishing Your O.T.P")
```
Comment (nednaZ) [inline code/modules/jobs/job_types/chaplain.dm:107]: Seems out of scope for this PR.

File: code/modules/judaism/foreskin.dm [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	if (M)
+		if (M.gender == FEMALE)
+			qdel(src)
+
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		foreskincolor = "#" + skintone2hex(H.skin_tone)
+	add_atom_colour(foreskincolor, FIXED_COLOUR_PRIORITY)
+
+/obj/item/organ/foreskin/prepare_eat()
```
Comment (nednaZ) [inline code/modules/judaism/foreskin.dm:22]: Deep frying the foreskin should have a special effect.

File: code/modules/judaism/foreskin.dm [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	S.w_class = w_class
+	S.add_atom_colour(foreskincolor, FIXED_COLOUR_PRIORITY)
+
+	return S
+
+/datum/surgery_step/incise/circumcise
+	name = "circumcise"
+	var/obj/item/organ/foreskin/F = null
+
+/datum/surgery/circumcision
```
Comment (nednaZ) [inline code/modules/judaism/foreskin.dm:37]: I'd suggest a "uncircumcise" surgery to revert the effect, Make it require some heavy weights as a surgical tool to stretch the remaining skin.

File: code/modules/judaism/foreskin.dm [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	requires_bodypart_type = 0
+
+/datum/surgery_step/incise/circumcise/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	return ..()
+
+/datum/surgery/circumcision/can_start(mob/user, mob/living/carbon/target)
+	var/obj/item/organ/foreskin/F = target.getorganslot(ORGAN_SLOT_FORESKIN)
+
+	if (target.gender == FEMALE)
+		to_chat(user, "<span class='warning'>We're not barbarians.</span>")
```
Comment (nednaZ) [inline code/modules/judaism/foreskin.dm:51]: It's 2522, it's the current year. Let surgeons do what they want.

File: code/modules/judaism/rabbi.dm [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+
+/obj/effect/landmark/start/rabbi
+	name = "Rabbi"
+	icon = 'ppstation/icons/jew.dmi'
+	icon_state = "jewspawn"
+
+/datum/job/rabbi
+	title = "Rabbi"
+	flag = RABBI
```
Comment (nednaZ) [inline code/modules/judaism/rabbi.dm:48]: Looks like this flag hasn't been defined.

File: code/modules/judaism/rabbi.dm [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	selection_color = "#dddddd"
+	access = list(ACCESS_SYNAGOGUE, ACCESS_MAINT_TUNNELS)
+	minimal_access = list(ACCESS_SYNAGOGUE, ACCESS_MAINT_TUNNELS)
+	outfit = /datum/outfit/job/rabbi
+	paycheck = PAYCHECK_COMMAND
+	paycheck_department = ACCOUNT_CIV
+
+/datum/job/rabbi/after_spawn(mob/living/carbon/human/H, mob/M)
+
+	var/lastname = pick(GLOB.last_names)
```
Comment (nednaZ) [inline code/modules/judaism/rabbi.dm:63]: Take from the last name of the player rather than generate a new one?

File: code/modules/judaism/rabbi.dm [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+	var/jewname = "Rabbi [lastname]stein"
+	H.fully_replace_character_name(H.real_name, jewname)
+	var/obj/item/organ/foreskin/F = H.getorgan(/obj/item/organ/foreskin)
+	if(F)
+		qdel(F)
+	if(H.gender == FEMALE)
+		H.gender = MALE
+		H.regenerate_icons()
+	var/datum/martial_art/S = H.mind.has_martialart("Catfighting")
```
Comment (nednaZ) [inline code/modules/judaism/rabbi.dm:72]: Perhaps give them CQC in the synagogue as a means of "Jewjitsu"?

File: code/modules/judaism/rabbi.dm [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+/datum/outfit/job/rabbi
+	name = "Rabbi"
+	jobtype = /datum/job/rabbi
+	uniform =  /obj/item/clothing/under/lawyer/black/rabbi
+	shoes = /obj/item/clothing/shoes/laceup
+	suit = /obj/item/clothing/suit/toggle/lawyer/black/rabbi
+	head = /obj/item/clothing/head/pp/jhat
+	neck = /obj/item/clothing/neck/necklace/dope/rabbi
+
+/obj/item/book/manual/talmud
```
Comment (nednaZ) [inline code/modules/judaism/rabbi.dm:86]: This could be a subtype of the bible to give them a form of blessing water.

File: code/modules/judaism/sacrifice.dm [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+	icon = 'ppstation/icons/jew.dmi'
+	icon_state = "jewaltar"
+	density = TRUE
+	anchored = TRUE
+	layer = TABLE_LAYER
+	climbable = FALSE
+	pass_flags = LETPASSTHROW //You can throw objects over this, despite it's density.")
+	max_integrity = 10000
+	integrity_failure = 30
+	var/foreskincount = 0
```
Comment (nednaZ) [inline code/modules/judaism/sacrifice.dm:20]: This variable likely doesn't need to exist.
If it's just for an error message, you can create a check on if any foreskins are found on the altar.

File: code/modules/judaism/sacrifice.dm [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+		to_chat(user, "<span class='notice'>The altar lights up...</span>")
+		sleep(10)
+		for (var/obj/item/organ/foreskin/M in get_turf(src))
+			sleep(10)
+			playsound(src, "sound/effects/splat.ogg", 30, 1, 5)
+			qdel(M)
+			var/obj/item/stack/spacecash/c1000/dosh = new/obj/item/stack/spacecash/c1000(src.loc)
+			dosh.throw_at(user, 1, 1)
+			foreskincount += 1
+		if(foreskincount == 0)
```
Comment (nednaZ) [inline code/modules/judaism/sacrifice.dm:44]: I'd suggest making this if(!M) and putting it within the for loop so then the error doesn't require a separate variable.

File: code/modules/judaism/sacrifice.dm [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+			to_chat(user, "<span class='notice'>...but nothing happens.</span>")
+		icon_state = "jewaltar"
+		foreskincount = 0
+	else
+		to_chat(user, "<span class='warning'>The altar burns you, goy!</span>")
+		user.apply_damage(10,"burn","chest")
+		user.say("*scream")
+		icon_state = "jewaltar_goyim"
+		sleep(10)
```
Comment (nednaZ) [inline code/modules/judaism/sacrifice.dm:53]: Rapidly tapping the altar won't burn the target, could have a variable with a world.time until next burn.

**Comment by ghost** [comment]
Thanks for the code review, Nednaz. 

I'd appreciate credit, Acensti, because I wrote all this code for PP Station 13, aside from the VERY lame chaplain.dm change.

**Comment by Acensti** [comment]
thanks for the cock content

**Comment by ghost** [comment]
Pretty sure you didn't read the code before copypasting it, some of the things are there to deal with specific PP13 features :flushed:
